cmd/thanos/query.go

```diff
@@ -398,6 +403,12 @@ func runQuery(
 		dns.ResolverType(dnsSDResolver),
 	)
 
+	for _, endpoint := range strictEndpoints {
+		if dns.IsDynamicNode(endpoint) {
+			return errors.Errorf("%s is a dynamically specified endpoint i.e. it uses SD and that is not permitted under strict mode. Use --endpoint for this", endpoint)
+		}
+	}
+
```

Comment:
Nit: Having this right after doing the same for `strictStores` would read tiny bit better :blush: